src/js/data.js

```diff
@@ -270,6 +269,17 @@ DataFactory.prototype = {
   }
 }
 
+var DataError = function(message) {
```

Comment:
Here's another case where, if it's widely done differently in JS, I'm happy to learn that, but in languages where I have more experience, I would call "data.js" a module, and "DataFactory" a class in that module. Having an Error class as an "inner class" on DataFactory seems unhelpfully complex, especially since it is also useful to export DataError to support the unit tests.